streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -286,48 +314,144 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated, f
                     // ignore
                 }
             }
+            return createdTasks;
         }
 
-        abstract void createTask(final TaskId id, final Set<TopicPartition> partitions);
+        abstract Task createTask(final Consumer<byte[], byte[]> consumer, final TaskId id, final Set<TopicPartition> partitions);
+
+        public void close() {}
     }
 
-    class TaskCreator extends AbstractTaskCreator {
+    static class TaskCreator extends AbstractTaskCreator {
+        private final ThreadCache cache;
+        private final KafkaClientSupplier clientSupplier;
+        private final String threadClientId;
+        private final Producer<byte[], byte[]> threadProducer;
+
+
+        TaskCreator(final InternalTopologyBuilder builder,
+                    final StreamsConfig config,
+                    final StreamsMetrics streamsMetrics,
+                    final StateDirectory stateDirectory,
+                    final Sensor taskCreatedSensor,
+                    final ChangelogReader storeChangelogReader,
+                    final ThreadCache cache,
+                    final Time time,
+                    final KafkaClientSupplier clientSupplier,
+                    final Producer<byte[], byte[]> threadProducer,
+                    final String threadClientId,
+                    final long rebalanceTimeoutMs,
+                    final String logPrefix) {
+            super(
+                    builder,
+                  config,
+                  streamsMetrics,
+                  stateDirectory,
+                  taskCreatedSensor,
+                  storeChangelogReader,
+                  time,
+                  rebalanceTimeoutMs,
+                  logPrefix);
+            this.cache = cache;
+            this.clientSupplier = clientSupplier;
+            this.threadProducer = threadProducer;
+            this.threadClientId = threadClientId;
+        }
+
         @Override
-        void createTask(final TaskId taskId, final Set<TopicPartition> partitions) {
-            final StreamTask task = createStreamTask(taskId, partitions);
+        StreamTask createTask(final Consumer<byte[], byte[]> consumer, final TaskId taskId, final Set<TopicPartition> partitions) {
+            taskCreatedSensor.record();
+            try {
+                return new StreamTask(
+                        taskId,
+                        applicationId,
+                        partitions,
+                        builder.build(taskId.topicGroupId),
+                        consumer,
+                        storeChangelogReader,
+                        config,
+                        streamsMetrics,
+                        stateDirectory,
+                        cache,
+                        time,
+                        createProducer(taskId));
+            } finally {
+                log.trace("{} Created active task {} with assigned partitions {}", logPrefix, taskId, partitions);
```

Comment:
This log entry could be misleading, that even if there is an exception happened and hence no task created, it will still print as `created ...`; and in practice I have once encountered this issue before which affected the trouble shooting process, I think we should try to piggy-back fix it.